ui/src/plugins/dev.perfetto.PinAndroidPerfMetrics/handlers/pinBlockingCall.ts

```diff
@@ -103,6 +115,70 @@ class BlockingCallMetricHandler implements MetricHandler {
       title: trackName,
     };
   }
+
+  private async getFrameIdWithMaxDurationBlockingCall(
+    ctx: Trace,
+    metricData: BlockingCallMetricData,
+  ): Promise<QueryResult> {
+    const cuj = metricData.cujName;
+    const processName = metricData.process;
+    const blockingCallName = metricData.blockingCallName;
+
+    // Fetch the frame_id of the frame with the max duration blocking call.
+    return ctx.engine.query(`
+      INCLUDE PERFETTO MODULE android.frame_blocking_calls.blocking_calls_aggregation;
+
+      SELECT
+        frame_id
+      FROM _blocking_calls_frame_cuj
+      WHERE
+        process_name = '${processName}'
+        AND name = '${blockingCallName}'
+        AND cuj_name = '${cuj}'
+      -- select frame_id for the metric with the maximum duration.
+      ORDER BY dur DESC
+      LIMIT 1`);
+  }
+  private async frameWithMaxDurBlockingCallTrackConfig(
+    ctx: Trace,
+    metricData: BlockingCallMetricData,
+  ): Promise<
+    Pick<DebugSliceTrackArgs, 'data' | 'columns' | 'argColumns' | 'title'>
+  > {
+    let row: Row = {
+      frame_id: null,
+    };
+
+    try {
+      row = (
+        await this.getFrameIdWithMaxDurationBlockingCall(ctx, metricData)
+      ).firstRow({frame_id: LONG});
+    } catch (e) {
+      throw new Error(
+        `${e.message} caused by: No frame found for the given process, CUJ and blocking call.`,
```

Comment:
probably you wanted to reverse the order of this string?  like `No frame found for the given process, CUJ and blocking call. Caused by: ...`